### ISSUE TITLE
Regarding the behavior of max_seq_length in SFTTrainer

### ISSUE THREAD

**Issue by Taiki-azrs** [huggingface/trl#2400]
The SFTTrainer documentation states:

> SFTTrainer always pads by default the sequences to the max_seq_length argument of the SFTTrainer.
https://huggingface.co/docs/trl/main/en/sft_trainer#best-practices

However, when looking at the actual code, the Tokenizer appears to have padding=False, and it does not seem to pad sequences to the max_seq_length value.
https://github.com/huggingface/trl/blob/main/trl/trainer/sft_trainer.py#L420

How does SFTTrainer ensure that sequences are padded to the max_seq_length value?

**Comment by qgallouedec** [comment]
You're right, the documentation is wrong. Would you like to contribute by correcting it?

**Comment by umbilnm** [comment]
Hi! I can correct it. Based on the discussion, it seems we could take one of two approaches:
1) Completely remove this mention from the “Best Practices” section
2) Update the text to clarify that truncation (rather than padding) happens by default.
Could you let me know which approach is better?

**Comment by qgallouedec** [comment]
Probably the 2. What do you think?

**Comment by umbilnm** [comment]
Ok, also if max_seq_len isn't specified the trainer it sets to `min(1024, tokenizer.model_max_lenght)` (not 2048), so revised text may look like this:

> SFTTrainer truncates sequences by default to the max_seq_length specified. If max_seq_length is not provided, the trainer sets it to the minimum of tokenizer.model_max_length and 1024. Ensure you verify this setting before training to avoid unintended behavior.
